freqtrade/data/dataprovider.py

```diff
@@ -50,21 +56,27 @@ def _set_cached_df(self, pair: str, timeframe: str, dataframe: DataFrame) -> Non
         :param timeframe: Timeframe to get data for
         :param dataframe: analyzed dataframe
         """
-        self.__cached_pairs[(pair, timeframe)] = (dataframe, datetime.now(timezone.utc))
+        self.__cached_pairs[(pair, timeframe, candle_type)] = (
+            dataframe, datetime.now(timezone.utc))
 
     def add_pairlisthandler(self, pairlists) -> None:
         """
         Allow adding pairlisthandler after initialization
         """
         self._pairlists = pairlists
 
-    def historic_ohlcv(self, pair: str, timeframe: str = None) -> DataFrame:
+    def historic_ohlcv(
+        self,
+        pair: str,
+        timeframe: str = None,
+        candle_type: str = ''
+    ) -> DataFrame:
         """
         Get stored historical candle (OHLCV) data
         :param pair: pair to get the data for
         :param timeframe: timeframe to get data for
```

Comment:
as dataprovider is a user facing interface, i'd want to have both the docstring here, as well as the corresponding documentation updated

** this is valid for all public methods (methods not prefixed with `_`).